samples/monte-carlo-pi/monte-carlo-pi.cpp

```diff
@@ -0,0 +1,255 @@
+          std::cerr << e.what() << "\n";
+        }
+      }
+    });
+
+    // Get device and display information: name and platform
+    auto device = queue.get_device();
+    std::cout << "Selected " << device.get_info<sycl::info::device::name>()
+              << " on platform "
+              << device.get_info<sycl::info::device::platform>()
+                     .get_info<sycl::info::platform::name>()
+              << "\n";
+
+    // Define the SYCL program and kernel
+    auto context = queue.get_context();
+    sycl::program program(context);
+    // Build the Monte-Carlo Pi kernel
+    program.build_with_kernel_type<monte_carlo_pi_kernel>();
+    sycl::kernel kernel = program.get_kernel<monte_carlo_pi_kernel>();
+
+    /* If the desired work-group size doesn't satisfy the device, define a
+     * perfect/max work-group depending on the selected device and kernel
+     * maximum size allowance */
+    work_group_size = get_best_work_group_size(work_group_size, device, kernel);
+
+    /* Size of the total sums that are going to be stored in the results vector
+     * is set based on the defined work-group size */
+    results.resize(iterations / work_group_size);
+
+    // Allocate device memory
+    sycl::buffer<sycl::cl_float2> points_buf(points.data(),
+                                             sycl::range<1>(iterations));
+    sycl::buffer<sycl::cl_int> results_buf(
+        results.data(), sycl::range<1>(iterations / work_group_size));
+
+    queue.submit([&](sycl::handler& cgh) {
+      const size_t global_size = iterations;
+      const size_t local_size = work_group_size;
+
+      // Get access to the data (points and results) on the device
+      auto points_ptr =
+          points_buf.get_access<sycl::access::mode::read,
+                                sycl::access::target::constant_buffer>(cgh);
+      auto results_ptr = results_buf.get_access<sycl::access::mode::write>(cgh);
+      // Allocate local memory on the device (to compute results)
+      sycl::accessor<sycl::cl_int, 1, sycl::access::mode::write,
+                     sycl::access::target::local>
+          local_results_ptr(sycl::range<1>(local_size), cgh);
+
+      // Run the kernel
+      cgh.parallel_for(
+          program.get_kernel<monte_carlo_pi_kernel>(),
+          sycl::nd_range<1>(sycl::range<1>(global_size),
+                            sycl::range<1>(local_size)),
+          monte_carlo_pi_kernel(points_ptr, results_ptr, local_results_ptr));
+    });
+  } catch (const sycl::exception& e) {
+    std::cerr << "SYCL exception caught: " << e.what();
+    return 1;
+  } catch (const std::exception& e) {
+    std::cerr << "C++ exception caught: " << e.what();
+    return 2;
+  }
+
+  // Sum the results (auto copied back to host)
+  int in_circle = 0;
+  for (auto& result : results) {
+    in_circle += result;
+  }
+
+  // Calculate the final result of "pi"
+  float pi =
+      (4.0f * static_cast<float>(in_circle)) / static_cast<float>(iterations);
```

Comment:
You should be able to remove the static casts here since at least one of the operands is a float.